lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -53,13 +53,14 @@ function _storeInMDandDeleteData(bucketName, dataGetInfo, cipherBundle,
  * credentialScope (to be used for streaming v4 auth if applicable)
  * @param {RequestLogger} log - logger instance
  * @param {function} callback - callback function
+ * @param {string} [originOp=s3:ObjectRemoved:DeleteMarkerCreated] - origin operation
  * @return {undefined} and call callback with (err, result) -
  * result.contentMD5 - content md5 of new object or version
  * result.versionId - unencrypted versionId returned by metadata
  */
 function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         canonicalID, cipherBundle, request, isDeleteMarker, streamingV4Params,
-        log, callback) {
+        log, callback, originOp = 's3:ObjectRemoved:DeleteMarkerCreated') {
```

Comment:
From an API/doc point of view this looks odd, the `createAndStoreObject` origin op must be a PUT or something per se. I understand the need to pass it for the delete marker case but it should be named differently when passed as an option I think to make it clearer, maybe `deleteMarkerOriginOp`?

Reply:
https://github.com/scality/cloudserver/pull/5087/files#diff-5900fcd2b08c9d289eafb9136636f13e87709614176c5dc541b08039c5687fccL159

As you can see here, it was already `s3:ObjectRemoved:DeleteMarkerCreated` without any possibility of configuring it... I agree that this is a bit odd but this is probably out of scope of this ticket to change that here..

Reply:
I updated that to use a boolean as parameter instead of an hardcoded originOp, might be simpler now

Reply:
i fear this is still hackish... The issue is that `createAndStoreObject` can both create object (and set originOp automatically to `s3:ObjectCreated:Put`) or create delete marker, and this flag affects only the second case.

IMHO it seems the clean way would be to pass originOp indeed (with no default value), use it in both cases (both when creating delete marker and when creating version) and modify the 4 calls to specify the expected `originOp` everytime